src/Package/Impl/DataInspect/Office/CsvAppFileIO.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using Microsoft.Common.Core;
+using Microsoft.R.Components.ContentTypes;
+using Microsoft.R.Components.InteractiveWorkflow;
+using Microsoft.VisualStudio.R.Package.Shell;
+using Microsoft.VisualStudio.R.Package.Utilities;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Utilities;
+using static System.FormattableString;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.R.Package.DataInspect.Office {
+    internal static class CsvAppFileIO {
+        private static int _busy;
+
+        public static async Task OpenDataCsvApp(string expression) {
+            if (Interlocked.Exchange(ref _busy, 1) > 0) {
+                return;
+            }
+
+            var workflowProvider = VsAppShell.Current.ExportProvider.GetExportedValue<IRInteractiveWorkflowProvider>();
+            var workflow = workflowProvider.GetOrCreate();
+            var session = workflow.RSession;
+
+            var folder = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
+            folder = Path.Combine(folder, @"RTVS_CSV_Exports\");
+            if(!Directory.Exists(folder)) {
+                Directory.CreateDirectory(folder);
+            }
+
+            // Strip "base::.GlobalEnv$"
```

Comment:
If you pass `DebugEvaluationResult` here instead of `string`, you can use its `Name` property to get the tail instead of stripping the prefix (and it'll then work in any frame, not just global - once JP commits packages, you'll have the same problem with them).